qt/python/mantidqt/project/project.py

```diff
@@ -0,0 +1,138 @@
+
+
+class Project(AnalysisDataServiceObserver):
+    def __init__(self):
+        super(Project, self).__init__()
+        # Has the project been saved, to Access this call .saved
+        self.__saved = True
+
+        # Last save locations
+        self.last_project_location = None
+
+        self.observeAll(True)
+
+        self.project_file_ext = ".mtdproj"
+
+    def save(self):
+        if self.last_project_location is None:
+            self.save_as()
+        else:
+            # Clear unused workspaces
+            self._clear_unused_workspaces(self.last_project_location)
+            # Actually save
+            workspaces_to_save = AnalysisDataService.getObjectNames()
+            project_saver = ProjectSaver(self.project_file_ext)
+            project_saver.save_project(directory=self.last_project_location, workspace_to_save=workspaces_to_save,
+                                       interfaces_to_save=None)
+            self.__saved = True
+
+    def save_as(self):
+        directory = self._get_directory_finder(accept_mode=QFileDialog.AcceptSave)
+
+        # If none then the user cancelled
+        if directory is None:
+            return
+
+        # todo: get a list of workspaces but to be implemented on GUI implementation
+        self.last_project_location = directory
+        workspaces_to_save = AnalysisDataService.getObjectNames()
+        project_saver = ProjectSaver(self.project_file_ext)
+        project_saver.save_project(directory=directory, workspace_to_save=workspaces_to_save, interfaces_to_save=None)
+        self.__saved = True
+
+    @staticmethod
+    def _get_directory_finder(accept_mode):
+        directory = None
+        # Check if it exists
+        first_pass = True
+        while first_pass or (not os.path.exists(directory) and os.path.exists(directory + (os.path.basename(directory)
+                                                                                           + ".mtdproj"))):
+            first_pass = False
+            directory = open_a_file_dialog(accept_mode=accept_mode, file_mode=QFileDialog.DirectoryOnly)
+            if directory is None:
+                # Cancel close dialogs
+                return
+
+        return directory
+
+    def load(self):
+        directory = self._get_directory_finder(accept_mode=QFileDialog.AcceptOpen)
+
+        # If none then the user cancelled
+        if directory is None:
+            return
+
+        project_loader = ProjectLoader(self.project_file_ext)
+        project_loader.load_project(directory)
+        self.last_project_location = directory
+
+    def offer_save(self, parent):
+        """
+        :param parent: QWidget; Parent of the QMessageBox that is popped up
+        :return: Bool; Returns false if no save needed/save complete. Returns True if need to cancel closing.
+        """
+        # If the current project is saved then return and don't do anything
+        if self.__saved:
+            return
```

Comment:
This will return `None` but docs suggest we should be returning a boolean?

Reply:
Doc's adjusted, as this is a check to make sure that saved is set already.